pkg/controller/registry/resolver/instrumented_resolver_test.go

```diff
@@ -0,0 +1,55 @@
+package resolver
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/stretchr/testify/require"
+)
+
+var result float64
+
+const (
+	failure = 0.0
+	success = 1.0
+)
+
+type fakeResolverWithError struct{}
+type fakeResolverWithoutError struct{}
+
+func (r *fakeResolverWithError) ResolveSteps(namespace string, sourceQuerier SourceQuerier) ([]*v1alpha1.Step, []v1alpha1.BundleLookup, []*v1alpha1.Subscription, error) {
+	return nil, nil, nil, errors.New("Fake error")
+}
+
+func (r *fakeResolverWithoutError) ResolveSteps(namespace string, sourceQuerier SourceQuerier) ([]*v1alpha1.Step, []v1alpha1.BundleLookup, []*v1alpha1.Subscription, error) {
+	return nil, nil, nil, nil
+}
+
+func newFakeResolverWithError() *fakeResolverWithError {
+	return &fakeResolverWithError{}
+}
+
+func newFakeResolverWithoutError() *fakeResolverWithoutError {
+	return &fakeResolverWithoutError{}
+}
+
+func changeToFailure(num *float64) {
+	*num = failure
+	result = *num
+}
+
+func changeToSuccess(num *float64) {
+	*num = success
+	result = *num
+}
+
+func TestInstrumentedResolver(t *testing.T) {
+	instrumentedResolver := NewInstrumentedResolver(newFakeResolverWithError(), changeToSuccess, changeToFailure)
+	instrumentedResolver.ResolveSteps("", nil)
+	require.Equal(t, result, failure)
```

Comment:
Won't this assertion hold even if `changeToFailure` does nothing at all or is never called? Or if `changeToSuccess` is called followed by `changeToFailure`?